pkg/api/testing/fuzzer.go

```diff
@@ -361,9 +371,9 @@ func FuzzerFor(t *testing.T, version string, src rand.Source) *fuzz.Fuzzer {
 		},
 		func(s *extensions.HorizontalPodAutoscalerSpec, c fuzz.Continue) {
 			c.FuzzNoCustom(s) // fuzz self without calling this function again
-			minReplicas := c.Rand.Int()
+			minReplicas := int(c.Rand.Int31())
 			s.MinReplicas = &minReplicas
-			s.CPUUtilization = &extensions.CPUTargetUtilization{TargetPercentage: int(c.RandUint64())}
+			s.CPUUtilization = &extensions.CPUTargetUtilization{TargetPercentage: int(c.Int31())}
```

Comment:
Going to fix with int(int32(c.RandUint64()))

On Wed, Nov 18, 2015 at 6:22 PM, Daniel Smith <notifications@github.com>
wrote:

> In pkg/api/testing/fuzzer.go
> <https://github.com/kubernetes/kubernetes/pull/17353#discussion_r45276960>
> :
>
> >  			s.MinReplicas = &minReplicas
> > -			s.CPUUtilization = &extensions.CPUTargetUtilization{TargetPercentage: int(c.RandUint64())}
> > +			s.CPUUtilization = &extensions.CPUTargetUtilization{TargetPercentage: int(c.Int31())}
>
> This will no longer test negative numbers.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/17353/files#r45276960>.
>